File: services/order-service/src/main/java/com/workbeattalent/orderservice/payment/PaymentRequest.java
package com.workbeattalent.orderservice.payment;

import com.workbeattalent.orderservice.customer.CustomerResponse;
import com.workbeattalent.orderservice.order.PaymentMethod;

import java.math.BigDecimal;

public record PaymentRequest(
        Long id,
        BigDecimal amount,
        PaymentMethod paymentMethod,
        Long orderId,
        String orderRef,
        CustomerResponse customer
) {
}

File: services/payment-service/src/main/kotlin/com/workbeattalent/paymentservice/payment/dto/PaymentRequest.kt
package com.workbeattalent.paymentservice.payment.dto

import com.workbeattalent.paymentservice.payment.PaymentMethod
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import java.math.BigDecimal

data class PaymentRequest(
    val id: Long?,
    @Positive(message = "Amount requires a positive value")
    val amount: BigDecimal,
    val paymentMethod: PaymentMethod,
    @NotNull(message = "Order identifier required")
    val orderId: Long,
    @NotNull(message = "Order reference required")
    val orderRef: String,
    val customer: Customer
)
